clippy_lints/src/floating_point_arithmetic.rs

```diff
@@ -379,7 +379,7 @@ fn detect_hypot(cx: &LateContext<'_>, args: &[Expr<'_>]) -> Option<String> {
             if let Some((rvalue, _)) = constant(cx, cx.typeck_results(), rargs_1);
             if Int(2) == lvalue && Int(2) == rvalue;
             then {
-                return Some(format!("{}.hypot({})", Sugg::hir(cx, largs_0, ".."), Sugg::hir(cx, rargs_0, "..")));
+                return Some(format!("{}.hypot({})", Sugg::hir(cx, largs_0, "..").maybe_par(), Sugg::hir(cx, rargs_0, "..")));
```

Comment:
Since ``Sugg::BinOp`` no longer holds the entire snippet, we need to explicitly call ``maybe_par()``.

For example, ```x + 1.hypot(y)``` when it should be ```(x + 1).hypot(y)```.

Reply:
I see...the parentheses are not preserved in the HIR.